### DISCUSSION TITLE
TTL of system.queries

### DISCUSSION THREAD

**Discussion by ruests** [trinodb/trino#22732] [Q&A] [answered]
What period of time are the system.queries records stored for?  Is there a configurable TTL on that table?  The documentation says _"The queries table contains information about currently and recently running queries on the Trino cluster."_, and I'm wondering what "recently" means.

**Comment by hashhar** [comment, marked as answer]
https://trino.io/docs/current/admin/properties-query-management.html#query-max-history and https://trino.io/docs/current/admin/properties-query-management.html#query-min-expire-age control this.

Note that having a very large value for the history means many more objects are retained in coordinator memory leading to high memory utilisation on coordinator.

If you want a persistent query history consider using or writing an EventListener implementation - Trino ships with two - MySQL and HTTP Event Listeners.
